stew/objects.nim

```diff
@@ -17,3 +17,10 @@ template init*(lvalue: var auto, a1, a2, a3: auto) =
 when not declared(default):
   proc default*(T: type): T = discard
 
+template countof*[R, T](v: array[R, T]): int =
+  sizeof(v) div sizeof(T)
```

Comment:
`R` is from `range`. 

This is the default internal treatments that the compiler will apply to this "wildcard" variable and using `R` as a name reminds me of this (there are various subtle differences between ranges and static ints).

Reply:
For example, the reason why this template is using `sizeof` is because this allows it to support enun-based arrays (which must use the range representation).

Reply:
hm, I find that pretty odd - comment code pls?

Reply:
hold on, how is this different from `len`?

Reply:
Hah, silly indeed. I guess old habits die hard :)

Reply:
Addressed in 9ac177d